R/zchunk_LA100.IEA_downscale_ctry.R

```diff
@@ -0,0 +1,308 @@
+      L100.USSR_Yug_ctry_bal$`1990_share` <- NULL
+
+      # Composite regions where population is used to downscale energy to countries over all historical years
+      # Subset composite regions and repeat by number of countries in each (139-142)
+      filter(L100.IEAcomposite, COUNTRY == "Other Africa") %>%
+        repeat_add_columns(tibble(iso = IEA_ctry_composite$iso[IEA_ctry_composite$IEA_ctry == "Other Africa"])) ->
+        L100.Afr_repCtry
+      filter(L100.IEAcomposite, COUNTRY == "Other Non-OECD Americas") %>%
+        repeat_add_columns(tibble(iso = IEA_ctry_composite$iso[IEA_ctry_composite$IEA_ctry == "Other Non-OECD Americas"])) ->
+        L100.LAM_repCtry
+      filter(L100.IEAcomposite, COUNTRY == "Other Asia") %>%
+        repeat_add_columns(tibble(iso = IEA_ctry_composite$iso[IEA_ctry_composite$IEA_ctry == "Other Asia"])) ->
+        L100.Asia_repCtry
+
+      # Combine these into a single data table and calculate population shares (144-149)
+      L100.Others_repCtry <- bind_rows(L100.Afr_repCtry, L100.LAM_repCtry, L100.Asia_repCtry)
+      # Note L100.Pop_thous_ctry_Yh is in long format
+      L100.Pop_thous_ctry_Yh %>%
+        filter(iso %in% L100.Others_repCtry$iso) %>%
+        left_join_error_no_match(select(IEA_ctry_composite, iso, IEA_ctry), by = "iso") %>%
+        rename(IEAcomposite = IEA_ctry) ->
+        L100.Others_pop
+
+      # Aggregate by country-within-composite-region and year to calculate population shares (151-156)
+      L100.Others_pop %>%
+        filter(year %in% HISTORICAL_YEARS) %>%
+        group_by(IEAcomposite, year) %>%
+        summarise(composite_population = sum(value)) ->
+        L100.Composites_pop
+
+      L100.Others_pop %>%
+        filter(year %in% HISTORICAL_YEARS) %>%
+        left_join_error_no_match(L100.Composites_pop, by = c("IEAcomposite", "year")) %>%
+        mutate(pop_share = value / composite_population) %>%
+        # drop extraneous columns
+        select(iso, year, pop_share) ->
+        L100.Others_pop_share
+
+      # Spread L100.Others_pop_share to wide format in preparation for computing L100.Others_ctry_bal
+      # This is relatively inexpensive
+      L100.Others_pop_share <- spread(L100.Others_pop_share, year, pop_share)
+
+      # Multiply the repeated country databases by the population shares to get the energy balances by country (159-162)
+      L100.Others_repCtry %>%
+        select(iso, FLOW, PRODUCT, COUNTRY) %>%
+        left_join_error_no_match(L100.Others_repCtry, by = c("iso", "FLOW", "PRODUCT", "COUNTRY")) %>%
+        rename(IEAcomposite = COUNTRY) ->
+        L100.Others_ctry_bal
+
+      L100.Others_repCtry %>%
+        select(iso) %>%
+        left_join_error_no_match(L100.Others_pop_share, by = "iso") %>%
+        select(-iso) ->
+        pop_share
+
+      hyc <- which(names(L100.Others_ctry_bal) %in% HISTORICAL_YEARS)
```

Comment:
Same comment as in previous usage of `which`.  When we calculated `hy` (line 59) we didn't seem to need `which`.  Is there something different going on here?  If not, then we should do it the same way in both places.